pkg/isoeditor/stream.go

```diff
@@ -39,11 +41,24 @@ func NewRHCOSStreamReader(isoPath string, ignitionContent *IgnitionContent, ramd
 		}
 	}
 
+	if kargs != nil {
+		files, err := KargsFiles(isoPath)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to read files to patch for kernel arguments")
+		}
+		for _, file := range files {
+			r, err = kargsReaderForContent(isoPath, file, r, bytes.NewReader(kargs))
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to create overwrite reader for kernel arguments in file \"%s\"", file)
+			}
+		}
+	}
+
 	return r, nil
 }
 
-func readerForContent(isoPath string, filePath string, base io.ReadSeeker, contentReader *bytes.Reader) (overlay.OverlayReader, error) {
-	start, length, err := GetISOFileInfo(filePath, isoPath)
+func readerForContentWithBoundariesFinder(isoPath, filePath string, base io.ReadSeeker, contentReader *bytes.Reader, boundariesFinder BoundariesFinder) (overlay.OverlayReader, error) {
```

Comment:
I think just change `readerForContent` to be the generic version.
If you want to eliminate the `boundariesFinder` function here you could also find the start and length in the caller. That feels cleaner than passing in a function type to a helper.

Reply:
So every caller of this function will call it (and handle error from it) before calling `readerForContent`. Something like
```
    start, length, err := boundariesFinder(isoPath, filePath)
    if err != nil {
           return nil, nil, err
    }
   return readerForContent(start, length, base, contentRader)
```